### DISCUSSION TITLE
Tooltip component

### DISCUSSION THREAD

**Discussion by divine** [tailwindlabs/headlessui#436] [Ideas]
Hopefully this will be in a next release:

https://trello.com/c/KVOX3uCK/56-tooltips-popups

https://material-ui.com/components/tooltips/
https://vuetifyjs.com/en/components/tooltips/
https://vue.chakra-ui.com/tooltip
https://reakit.io/docs/tooltip/

________

Update: https://github.com/tailwindlabs/headlessui/discussions/436#discussioncomment-6632690

**Comment by ljcremer** [comment]
This would be great. Another resource to look at: 
[https://github.com/Akryum/v-tooltip](https://github.com/Akryum/v-tooltip)

**Comment by sebastibe** [comment]
I found this which seems to be based on headlessui: https://www.vechaiui.com/tooltip

Would be amazing to have something similar directly available from headlessui instead.

**Comment by divine** [comment]
Any update on this?

**Comment by meotimdihia** [comment]
@RobinMalfait Sorry for pinging you.

**Comment by Psycarlo** [comment]
Any update?

**Comment by divine** [comment]
@adamwathan sorry to bump you, this tooltip was in a TailwindUI Trello to-do list.

If you plan to add a tooltip, please just let us know. It's sad to use another library for showing only tooltip.

I hope it makes sense after 2 years.

Thanks!

**Comment by suryarajendhran** [comment]
Pinging here to keep the thread active. It would be really great to have the tooltip as part of headlessui

**Comment by DeamonDev** [comment]
Pinging it too...Main competitor which is BaseUI have planned to add tooltip recently.

**Comment by Psycarlo** [comment]
Pinging again. Would be awesome to see

**Comment by divine** [comment]
Tailwind's team doesn't seem to care about Headless UI anymore. It's been stuck without new components since the last update on July 7, 2021, and there have been no recent additions for two years.

Okay, that might have been a bit too harsh since it’s open source and do whatever you want, but what about TailwindUI customers who requested a tooltip? It was added to Trello and seemed to be one of the most requested features.

I remember when Tailwind Labs was small, and they fulfilled every customer request. Sadly, it seems that they have become too big to listen to customers.

Just my two cents.

**Reply by meotimdihia** [reply]
I think they intended to build in the catalyst template:

https://catalyst.tailwindui.com/docs

**Comment by imakeinternet** [comment]
Ping

**Reply by meotimdihia** [reply]
I think they intended to build in the catalyst template:

https://catalyst.tailwindui.com/docs

**Comment by meotimdihia** [comment]
I think they intended to build in the catalyst template:

https://catalyst.tailwindui.com/docs

**Reply by jacktsin1** [reply]
Although it's missing from there as well...

**Comment by bigperson** [comment]
Pinging here to keep the thread active.

**Comment by aschenkuttel** [comment]
+1

**Comment by Psycarlo** [comment]
Just use https://www.radix-vue.com

**Comment by itsk3vin** [comment]
pinging to keep this active--this would be great

**Comment by arif-js** [comment]
PING!

**Comment by tomastrivino** [comment]
ping

**Comment by meotimdihia** [comment]
We should just switch to `Shadcn`. 
I don't think headlessui team will do it. They haven't released a new component for 2 years.
Even `catalyst` is worse than `Shadcn` in its current state.

**Reply by imakeinternet** [reply]
I could not agree more with this statement.

**Reply by tomastrivino** [reply]
Regarding this, Shadcn recently released a new version including support for TailwindCSS V4.

**Reply by KeyboardRage** [reply]
Unfortunately Shadcn does not support Vue, from what I have seen

**Reply by Psycarlo** [reply]
> Unfortunately Shadcn does not support Vue, from what I have seen

shadcn-vue
reka-ui

**Reply by ivan-kleshnin** [reply]
ShadCn is the same mess 😞

**Comment by jonathansolisc** [comment]
I've got a workaround for this, but it only works for tooltips whose trigger does not have/need an `onClick` event. For instance it will show the tooltip on hover but It won't fire the click event for links.

```jsx
import { useRef } from "react";

export default function MyComponent(){
  const buttonRef = useRef(null);

  return (
  <Popover>
      {({ close }) => (
        <>
          <PopoverButton
            ref={buttonRef}
            className="cursor-pointer"
            onMouseEnter={() => buttonRef.current?.click()}
            onMouseLeave={() => close()}
          >
            Hover this to open tooltip
          </PopoverButton>

          <PopoverPanel
            anchor={position}
            className="ml-2 flex flex-col rounded-md bg-gray-900 px-2 py-1 text-sm font-medium text-white shadow-xl"
          >
            This is the content of my tooltip!
          </PopoverPanel>
        </>
      )}
    </Popover>
  )

}
```

**Comment by seansean11** [comment]
After searching the code, I see that they do have a [Tooltip component](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/components/tooltip/tooltip.tsx) built, but have it [commented out](https://github.com/tailwindlabs/headlessui/blob/main/packages/%40headlessui-react/src/index.ts#L25-L26) from being bundled.  Does anyone know why they have not released this?

**Reply by tomastrivino** [reply]
LOL

**Reply by adamwathan** [reply]
Mostly just need documentation, but also some API review and testing. We are using it on tailwindcss.com though 😄 Maybe we can release it early with no docs for anyone who wants to test it for now.

https://github.com/user-attachments/assets/ededb2fd-5f8c-4e67-a336-5179c6fc9841

**Reply by sc4lr** [reply]
That'd be great! I can't wait to try it!

**Reply by mrcoles** [reply]
This would be great to try out! I’m finding other React tooltip libraries extremely non-performant

**Reply by seansean11** [reply]
Please do!

**Reply by wtusfo** [reply]
@adamwathan its been 2 months already, dude

**Reply by arshad-yaseen** [reply]
@adamwathan please, also in catalyst

**Reply by czibbelvin** [reply]
I'd love to get my hands on it so I don't have to pull in another third-party library to solve this

**Reply by Vondry** [reply]
@RobinMalfait Would be awesome to have Tooltip component in Headless UI 👍

**Reply by ludwighogstrom** [reply]
Need this as well!

**Reply by Vondry** [reply]
@ludwighogstrom I am now using https://github.com/rezasohrabi/headless-tooltip and it works without any issues even with SSR :)

**Reply by ludwighogstrom** [reply]
> @ludwighogstrom I am now using https://github.com/rezasohrabi/headless-tooltip and it works without any issues even with SSR :)

Thanks for the suggestion! We generally try to limit the number of libraries we use and prefer well-established packages for long-term stability. For now, we’re hoping Headless UI will add features that cover common UI needs.

**Reply by suuf** [reply]
May I also encourage uncommenting the tooltip 😁

**Reply by Psycarlo** [reply]
Nobody use this lib anymore.

reka ui for vue
and base ui for react

**Comment by rezasohrabi** [comment]
I recently ran into the tooltip limitation in Headless UI as well.

Changing my entire UI stack wasn’t an option, so I built something simple:
🎯 [headless-tooltip](https://github.com/rezasohrabi/headless-tooltip) — a minimal, accessible, and unstyled tooltip component for React.

🧠 Keyboard-accessible and screen reader friendly

🎨 No styling opinions (perfect with Tailwind/DaisyUI)

📦 Tiny, dependency-free, Headless UI-inspired

I made it specifically for cases like this where you want a tooltip that “just works” with Headless UI-style patterns.

Hope it helps someone here! Feedback and contributions welcome 🙌
